app/controllers/api/v1/courses_controller.rb

```diff
@@ -0,0 +1,79 @@
+module Api
+  # index/create/destroy/course_detail actions are in used.
+  module V1
+    # course_detail action is used to create and save detail object.
+    class CoursesController < ApplicationController
+      before_action :set_course, only: %i[show update]
+
+      # GET /courses
+      def index
+        @courses = Course.all
+
+        if user_authorized?
+          render json: @courses
+        else
+          render json: { error: 'Invalid API token! User not found.' }, status: :unprocessable_entity
```

Comment:
I think this maybe can be a method. Since it's something you'd continuously do. You can pass type of message (error, success etc., message itself etc.) It'd make for cleaner code and won't have to repeat self as much

Reply:
thank you. I did refactor in the front end as pointed out by you but I got limited time to project presentation today, but I will allocate some time to refactor near future.